### DISCUSSION TITLE
Basic OpenGallery() Routine

### DISCUSSION THREAD

**Discussion by ghost** [yasirkula/UnityNativeGallery#184] [Ideas]
Hello. 
   I was wondering if you could add a routine to just open the gallery without any expectation to select/return an image/video. After saving a video from our app I want to take the user to the Gallery so they have access to the full list of edit/share options offered by their device. On iOS I'm seeing success using OpenURL(), but on Android it seems to require direct work with the intent for some devices. (still figuring it out how to get this working)

Thanks for considering it and great asset/plugin.

**Comment by yasirkula** [comment]
It's a very specific and, rare I must say, use-case that I'm currently not planning to support. To resolve your issue on Android, googling "android open gallery java" [should help](https://stackoverflow.com/a/23821227/2373034).

**Reply by ghost** [reply]
Thanks for considering. With an app that allows for in-app recording it makes sense for us (and possibly others) to use the editing and sharing tools users are already familiar with on their phone vs. recreating it. Opening the Gallery for them is just a convenience feature so they can work on it right after saving. 

Below is what we came up with for a quick fix in case anyone else finds it useful. No idea how long this will continue to be supported, but seems to be working well for us at this time. (Note: using OpenURL() on Android with "content://media/internal/images/media" fails on some LG devices)

```
#if UNITY_ANDROID && !UNITY_EDITOR
	try
	{	
		var unityPlayer = new AndroidJavaClass("com.unity3d.player.UnityPlayer");
		var activity = unityPlayer.GetStatic<AndroidJavaObject>("currentActivity");						
		var intent = new AndroidJavaObject("android.content.Intent", "android.intent.action.MAIN");
		intent.Call<AndroidJavaObject>("addCategory","android.intent.category.LAUNCHER");
		intent.Call<AndroidJavaObject>("addCategory","android.intent.category.APP_GALLERY");
		activity.Call("startActivity", intent);
	}
	catch (Exception e)
	{	mtLog.Error("File","cannot open gallery.",e.Message); 
		return false; }
#elif UNITY_IOS && !UNITY_EDITOR
	Application.OpenURL("photos-redirect://");
#else
	mtLog.Error("File","unsupported platform for gallery","how did you get here");
#endif
```

Thanks again for a great asset/plugin. It's a big help.
